pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -139,35 +139,74 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 	}
-	setObjectLabels(cr, &env.SmartRouter)
-	setObjectLabels(cr, &env.ProcessMigration)
+	setObjectLabels(cr, &env.SmartRouter, getFormattedComponentName(cr, constants.Smartrouter))
+	setObjectLabels(cr, &env.ProcessMigration, getFormattedComponentName(cr, constants.ProcessMigration))
+}
+
+func getConsoleName(cr *api.KieApp) string {
+	env := string(cr.Spec.Environment)
+	consoleName := ""
+	if strings.Contains(env, constants.RhdmPrefix) {
+		consoleName = constants.RhdmDecisionCentral
+	} else if strings.Contains(env, constants.RhpamPrefix) {
+
+		if strings.Contains(env, constants.Production) {
+			consoleName = constants.RhpamBusinessCentralMon
+		} else {
+			consoleName = constants.RhpamBusinessCentral
+		}
+	}
+	return consoleName
 }
 
 func getSubComponentTypeByImageName(imageName string) string {
-	retValue := "application"
+	retValue := constants.SUBCOMPONENT_TYPE_APP
 	if imageName == constants.RhpamSmartRouterImageName || imageName == constants.RhpamControllerImageName ||
 		imageName == constants.RhdmSmartRouterImageName || imageName == constants.RhdmControllerImageName {
 
-		retValue = "infrastructure"
+		retValue = constants.SUBCOMPONENT_TYPE_INFRA
 	}
 	return retValue
 }
 
-func setObjectLabels(cr *api.KieApp, object *api.CustomObject) {
+func setObjectLabels(cr *api.KieApp, object *api.CustomObject, subcomponent string) {
 	for index, obj := range object.DeploymentConfigs {
-		subcomponent, _, _ := GetImage(object.DeploymentConfigs[index].Spec.Template.Spec.Containers[0].Image)
 		object.DeploymentConfigs[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, subcomponent, getSubComponentTypeByImageName(subcomponent))
 	}
 	for index, obj := range object.StatefulSets {
-		subcomponent, _, _ := GetImage(object.StatefulSets[index].Spec.Template.Spec.Containers[0].Image)
 		object.StatefulSets[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, subcomponent, getSubComponentTypeByImageName(subcomponent))
 	}
+}
 
+func setObjectLabelsForServer(cr *api.KieApp, object *api.CustomObject) {
+	for index, obj := range object.DeploymentConfigs {
+		subcomponent := getFormattedComponentName(cr, constants.KieServerServicePrefix)
+		object.DeploymentConfigs[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, subcomponent, getSubComponentTypeByImageName(subcomponent))
+	}
+	for index, obj := range object.StatefulSets {
+		subcomponent := getFormattedComponentName(cr, constants.KieServerServicePrefix)
+		object.StatefulSets[index].Spec.Template.Labels = setLabels(cr, obj.Spec.Template.Labels, subcomponent, getSubComponentTypeByImageName(subcomponent))
+	}
+}
+
+func getFormattedComponentName(cr *api.KieApp, name string) string {
+	return getPrefixEnv(cr) + "-" + name + constants.RhelVersion
+}
+
+func getPrefixEnv(cr *api.KieApp) string {
+	prefix := ""
+	if strings.HasPrefix(string(cr.Status.Applied.Environment), constants.RhpamPrefix) {
+		prefix = constants.RhpamPrefix
+	}
```

Comment:
would it make more sense to set prefix to `constants.RhpamPrefix` by default, and only do the strings check for the rhdm prefix?